source/usb_hid.c

```diff
@@ -473,16 +714,25 @@ static int usb_hid_worker(void *arg)
 		return ret;
 	host_fd = ret;
 
-	ret = os_ioctl(host_fd, USBV5_IOCTL_GETVERSION, NULL, 0, ver, sizeof(ver));
-	if (ret < 0)
-		return ret;
-
-	/* We only support USBv5 for now */
-	if (ver[0] != 0x50001)
-		return IOS_EINVAL;
-
-	ret = os_ioctl_async(host_fd, USBV5_IOCTL_GETDEVICECHANGE, NULL, 0, device_change_devices,
+	ret = os_ioctl(host_fd, USBV4_IOCTL_GETVERSION, NULL, 0, NULL, 0);
+	if (ret == 0x40001) {
```

Comment:
I'm confused by this change: going from `USBV5_IOCTL_GETVERSION` to `USBV4_IOCTL_GETVERSION` is equivalent to changing the ioctl code from 0 to 6 -- in other words, it's as if we were now calling `USBV5_IOCTL_ATTACHFINISH`, which doesn't make much sense. Or am I missing something?

Reply:
If the IOS in question is using the v4 api, then you get a 0x040001 return otherwise you don't and you can assume v5 is in use

Reply:
Yes, but if you are using the v5 API, then this generates the wrong call (USBV5_IOCTL_ATTACHFINISH). It seems very strange that on different API versions the GETVERSION ioctl has different opcodes, it pretty much defeats the point of it, because you cannot call it reliably.

On the other hand, I see that libogc does the same thing as your code, so I guess it's fine?

Reply:
Yep, that's where I copied it from lol

Reply:
Tbh though I haven't really been doing much with fakemote these days, I've gone the brainslug route https://github.com/Santroller/santroller-bslug